Snippets/Snippets_6/Host/EndpointStartAndStop.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Snippets6.Host
+{
+    using System.Threading.Tasks;
+    using NServiceBus;
+
+    #region host-EndpointStartAndStop
+
+    class RunWhenEndpointStartsAndStops : IWantToRunWhenEndpointStartsAndStops
+    {
+        public async Task Start(IMessageSession session)
+        {
+            // perform startup logic
+        }
+
+        public async Task Stop(IMessageSession session)
```

Comment:
Inform another system about your shutdown, heartbeats, etc. That was the initial reasoning when we introduced it.